### ISSUE TITLE
[BUG] Can't use custom variants for tag and input

### ISSUE THREAD

**Issue by fundlg** [csandman/chakra-react-select#215]
### Description

Hi, author.

When I have custom variants for input I expected that library will allow me to use it inside select, but as I see in the source code - the usage is restricted due to the hardcoded list of variants. I expect that I can use my custom variants - it should be the same flow as with built-in variants so should work properly.

https://github.com/csandman/chakra-react-select/blob/cc51bbf997465d28cf1910f150753dfd15a366a7/src/use-chakra-select-props.ts#L89-L98

I would be thankful for any answer

### chakra-react-select Version

4.4.0

### TypeScript?

- [X] Yes I use TypeScript

### Steps to reproduce

1. Go to chakra-ui theme config
2. Create custom variant for input
3. Use it inside select `variant` prop

Expected - style variant is applied
Actual - style variant resets to the default one

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS/iPadOS
- [ ] Android

**Comment by csandman** [comment]
Currently this is the intended behavior to prevent styles from being applied strangely from unknown variants, but its possible that this could work alright.  Can you give me an example of a custom variant you made so I can see how it would work if applied?

**Comment by fundlg** [comment]
Sure. I have this object to define a variant special for select view - it should be different from other inputs in my case.
```
import { inputAnatomy } from '@chakra-ui/anatomy';
import { createMultiStyleConfigHelpers } from '@chakra-ui/react';

const { definePartsStyle } = createMultiStyleConfigHelpers(inputAnatomy.keys);

export const dropdown = definePartsStyle({
  field: {
    p: '20px',
    color: 'dark.card',

    fontWeight: 500,

    border: '1px solid',
    borderColor: 'light.dividers',
    borderRadius: '12px',

    backgroundColor: 'white.primary',

    _focus: {
      boxShadow: 'none',
    },

    _placeholder: {
      height: 'auto',
      color: 'dark.icons',
    },

    _disabled: {
      color: 'dark.text',
      backgroundColor: 'light.rows',
      borderColor: 'light.dividers',
    },

    _hover: {
      borderColor: 'yellow.200',

      _disabled: {
        color: 'dark.text',
        backgroundColor: 'light.rows',
        borderColor: 'light.dividers',
      },
    },

    _focusVisible: {
      borderColor: 'yellow.primary',
    },

    _active: {
      boxShadow: 'none',
      borderColor: 'yellow.primary',
    },

    _invalid: {
      color: 'red.primary',
      borderColor: 'red.primary',

      _disabled: {
        color: 'dark.text',
        backgroundColor: 'light.rows',
        borderColor: 'light.dividers',
      },
    },

    _dark: {
      borderColor: 'dark.dividers',
      backgroundColor: 'dark.buttons',

      _hover: {
        borderColor: 'yellow.primary',

        _disabled: {
          backgroundColor: 'dark.bg',
          borderColor: 'dark.dividers',
        },
      },

      _focus: {
        borderColor: 'yellow.primary',

        _disabled: {
          backgroundColor: 'dark.bg',
          borderColor: 'dark.dividers',
        },
      },

      _disabled: {
        backgroundColor: 'dark.bg',
        borderColor: 'dark.dividers',
      },
    },
  },
  element: {
    minH: '100%',
    color: 'dark.icons',

    mx: 4,
    boxSize: 4,

    display: 'flex',
    gap: 2,
  },
  addon: {
    backgroundColor: 'transparent',
  },
});
```
Then use `extendTheme` theme:
```
import {
  DeepPartial,
  extendTheme,
  theme as defaultTheme,
  ThemeOverride,
} from '@chakra-ui/react';

import colors from './overrides/colors';
import {Input} from './overrides/components/Input';

export const themeConfig: DeepPartial<ThemeOverride> = {
  config: {
    initialColorMode: 'dark',
  },
  styles: {
    global: {
      'html, body': {
        fontSize: '14px',
        fontFamily: 'Satoshi',
        lineHeight: '22px',
      },
    },
  },
  components: {
    ...defaultTheme.components,
    Input,
  },
  colors,
};

export const theme = extendTheme(themeConfig, defaultTheme);
```

I also use custom color scheme, so if you want to have those colors just add this object to the `colors` theme key:
```
{
  black: '#000000',

  white: {
    primary: '#FFFFFF',
    alpha: {
      20: 'rgba(255,255,255,0.2)',
      60: 'rgba(255,255,255,0.6)',
    },
  },

  yellow: {
    50: '#FFFDF6',
    100: '#FFF8E1',
    200: '#FFECB3',
    400: '#FFE082',
    600: '#FFB300',
    700: '#ED8E20',

    primary: '#FFCA28',

    alpha: {
      5: 'rgba(255, 202, 40, 0.05)',
      10: 'rgba(255, 202, 40, 0.1)',
    },
  },
  red: {
    50: '#FFF6F5',
    100: '#FFECEC',
    200: '#FEDAD9',
    400: '#FDB4B3',
    600: '#D32F2F',
    700: '#B71C1C',

    primary: '#FA4440',

    alpha: {
      5: 'rgba(250, 68, 64, 0.05)',
      10: 'rgba(250, 68, 64, 0.1)',
    },
  },

  light: {
    bg: '#F3F5F8',
    rows: '#FBFBFD',
    hover: '#F5F7FA',
    dividers: '#ECEDF3',
    innercard: '#F8F9FC',
    darkdividers: '#E2E5EA',
  },
  dark: {
    bg: '#0A0C0F',
    bar: '#0F1115',
    card: '#111317',
    rows: '#13151A',
    text: '#ADB4BD',
    icons: '#67707B',
    buttons: '#15191E',
    tooltip: '#1D2129',
    dividers: '#2B2F37',
    innercard: '#191D24',
    darkdividers: '#15181C',
  },
};
```

**Comment by csandman** [comment]
I'm sorry it took me so long to get around to this, I was pretty busy around when you made it, and it completely slipped my mind that it existed. I gave it a shot, allowing custom variants for both the tag and input, and I just published the changes in `v4.5.0`. Here's an example using the custom theme variant that you provided: https://codesandbox.io/s/qxmvmj?file=/app.tsx

There is one thing to bear in mind, though; the theme styles do not get the highest priority in terms of overriding the component's styles. After pulling styles from the Input's component theme, in multiple cases this package overrides them if they need to be changed in order for them to work in the context of a `Select` component. So your mileage may vary. If you still have any interest in this feature and give this a shot, let me know how it goes, or if you run into any issues making it work.
